resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolver.java

```diff
@@ -905,15 +912,24 @@ private boolean doResolveCached(String hostname,
                 for (int i = 0; i < numEntries; i++) {
                     final DnsCacheEntry e = cachedEntries.get(i);
                     if (f.addressType().isInstance(e.address())) {
+                        boolean entryExpired = false;
+                        if (e instanceof ExpirableDnsCacheEntry) {
+                            ExpirableDnsCacheEntry ee = (ExpirableDnsCacheEntry) e;
+                            long now = System.currentTimeMillis();
```

Comment:
I think it would be better to use `System.nanoTime()` to not be effected by time changes